File: 2024_04/0409/middle/minimumsum.java
class Solution {
    public int minimumSum(int num) {
        String numStr = Integer.toString(num);
        char[] numArray = numStr.toCharArray();
        
        for (int i = 0; i < 3; i++) {
            for (int j = i + 1; j < 4; j++) {
                if (numArray[i] > numArray[j]) {
                    char temp = numArray[i];
                    numArray[i] = numArray[j];
                    numArray[j] = temp;
                }
            }
        }
        
        int a = (numArray[0] - '0') * 10 + (numArray[2] - '0');
        int b = (numArray[1] - '0') * 10 + (numArray[3] - '0');
        
        return a + b;
    }
}

File: 2024_04/0409/middle/minimumsum.cpp
class Solution {
public:
    int minimumSum(int num) {
    char num_str[5];
    sprintf(num_str, "%d", num);
    
    char num_list[4];
    for (int i = 0; i < 4; i++) {
        num_list[i] = num_str[i];
    }
    
    for (int i = 0; i < 3; i++) {
        for (int j = i + 1; j < 4; j++) {
            if (num_list[i] > num_list[j]) {
                char temp = num_list[i];
                num_list[i] = num_list[j];
                num_list[j] = temp;
            }
        }
    }
    
    int a = (num_list[0] - '0') * 10 + (num_list[2] - '0');
    int b = (num_list[1] - '0') * 10 + (num_list[3] - '0');
    
    return a + b;
    }
};
